controllers/managedocs_controller.go

```diff
@@ -1618,10 +1629,17 @@ func (r *ManagedOCSReconciler) removeOLMComponents() error {
 
 	r.Log.Info("deleting deployer csv")
 	if csv, err := r.getCSVByPrefix(deployerCSVPrefix); err == nil {
-		if err := r.delete(csv); err != nil {
-			return fmt.Errorf("Unable to delete csv: %v", err)
+		// There's a possibility that csv will be deleted in previous reconcile and
+		// getCSVByPrefix will return emtpy csv in those cases, so need to check for value
+		if csv != nil {
```

Comment:
Same as the comment above about `getCSVByPrefix`

Reply:
same as above